### ISSUE TITLE
Functions not found after migrating to model v.4

### ISSUE THREAD

**Issue by petyakapchina** [Azure/azure-functions-nodejs-library#254]
Hello, I just switched my existing project to the new programming model v4 using [this migration guide](https://learn.microsoft.com/en-us/azure/azure-functions/functions-node-upgrade-v4?tabs=v4&pivots=programming-language-javascript).

#### Related information 
My folder structure:
```
.root
+---get-asset
|   |   index.ts
```
package.json
` "main": "dist/*/index.js"`
After complications, dist looks as follows:
![image](https://github.com/Azure/azure-functions-nodejs-library/assets/16178821/32449e42-5510-40da-8bc2-709c914bc687)

tsconfig.json
`"outDir": "dist"`

 Functions look like:
```
export async function getAsset(req: HttpRequest, context: InvocationContext): Promise<HttpResponseInit> {
...
return {
            ...
        }
}

app.http('get-asset', {
  methods: ['GET'],
  authLevel: 'anonymous',
  route: 'assets/{businessLine:alpha:length(2)}/{*path}',
  handler: getAsset,
});
```

I successfully run the functions on my local environment and am able to execute them:
```
Functions:

        get-asset: [GET] http://localhost:7071/assets/{businessLine:alpha:length(2)}/{*path}
```

However, when I try to deploy them on Azure, no functions are registered and I see the following log:
`No job functions found. Try making your job classes and methods public. If you're using binding extensions (e.g. Azure Storage, ServiceBus, Timers, etc.) make sure you've called the registration method for the extension(s) in your startup code (e.g. builder.AddAzureStorage(), builder.AddServiceBus(), builder.AddTimers(), etc.).`

I followed the [troubleshoot guide](https://learn.microsoft.com/en-us/azure/azure-functions/functions-node-troubleshoot?pivots=nodejs-model-v4)

Make sure you're using [Azure Functions Runtime Version](https://learn.microsoft.com/en-us/azure/azure-functions/functions-versions) 4.25 or higher.
<img width="294" alt="image" src="https://github.com/Azure/azure-functions-nodejs-library/assets/16178821/c6089d33-3186-4523-9242-8f2ad77e6e62">

Make sure you're using Node.js v18 or higher.
<img width="556" alt="image" src="https://github.com/Azure/azure-functions-nodejs-library/assets/16178821/dc20c0bb-e9c3-48b2-8652-70f0df029ca6">

I tried several things, like:
-> Using node.js v20
-> restructuring my project to put the files in src/functions with the respectful change in package.json => it aways worked locally and never in Azure
-> add AzureWebJobsFeatureFlags set to EnableWorkerIndexing in my functions config as suggested in [this issue](https://github.com/Azure/azure-functions-nodejs-library/issues/78)

#### Investigative information

Please provide the following:

- Timestamp: 07/05/2024, 13:02:42.028 UTC
- Function name: get-asset
- Region: West Europe

#### Expected behaviour 

To be able to successfully deploy the functions to Azure as they are running locally

#### Actual behaviour
When deploying the project, no functions are found in the cloud.

**Comment by ejizba** [comment]
Hi @petyakapchina, glad you found the troubleshooting guide! Did you by chance look for errors in your app insights? And did you see any?

Unfortunately the function name by itself isn't enough for us to investigate on our end. We need the app name, which you can either share here or follow [these instructions](https://github.com/Azure/azure-functions-host/wiki/Sharing-Your-Function-App-name-privately) to share privately.

**Comment by petyakapchina** [comment]
Hi @ejizba ,
Thanks for your fast response!

Yes, I ran the Insight query from the troubleshoot guide, after setting FUNCTIONS_NODE_BLOCK_ON_ENTRY_POINT_ERROR to true in the function settings, but did not see logs, that contain `where message has "entry point"`

I only got the same error, that I see in the Log analytics workspace, from the `traces` table: 
`No job functions found. Try making your job classes and methods public. If you're using binding extensions (e.g. Azure Storage, ServiceBus, Timers, etc.) make sure you've called the registration method for the extension(s) in your startup code (e.g. builder.AddAzureStorage(), builder.AddServiceBus(), builder.AddTimers(), etc.).`

I am deploying using the command `func azure functionapp publish <func-name> --typescript`

The `No job functions found...` error I am seeing is in the log analytics workspace from the FunctionAppLogs table as well.

This is the ID that I see in my logs:
```
[Information] Executed 'Functions.get-asset' (Succeeded, Id=f7955b3e-7c34-40e1-ba57-472cdbd6631d, Duration=133ms)
[Information] Executing 'Functions.get-asset' (Reason='This function was programmatically called via the host APIs.', Id=f7955b3e-7c34-40e1-ba57-472cdbd6631d)
```

Latest times stamp of my trying to deploy is at 08/05/2024, 13:03:37.637 UTC

I would also like to add that when I deploy with the `--list-included-files` option, I see all files being listed, inc. files from the .funcignore file
Mine looks like this:
```
**/*.js.map
**/*.ts
.git*
.vscode
.idea
local.settings.json
test
*.md
package.json
package-lock.json
tsconfig.json
yarn.lock
*.log
```

but I can see files like the package.json and the tsconfig being listed, not sure if this can be related to my issue.

**Comment by petyakapchina** [comment]
I found the issue, with the previous model v3, the path to the executable files was defined in the function.json file. Hence in our .funcignore file we had added the package.json.

However, when we migrated and move this definition in the package.json, it being omitted from the deploy, lead to this error.

After removing the package.json from the .funcigone, the functions appeared in the Azure portal as well.

Thanks for your support, @ejizba !
